chroma-common/blockdevices/blockdevice_zfs.py

```diff
@@ -129,8 +124,19 @@ def lock_pool(self):
         while not self.lock.i_am_locking():
             try:
                 self.lock.acquire(self.LOCK_ACQUIRE_TIMEOUT)
+                with open(self.lock.lock_file, 'w') as f:
+                    f.writelines(str(self.lock.pid))
             except LockTimeout:
-                pass
+                # prune locks owned by non-existent processes
+                with open(self.lock.lock_file, 'r') as f:
+                    contents = f.readlines()
+
+                assert len(contents) == 1 and contents[0].isdigit(), \
+                    'unexpected contents of lockfile %s: %s' % (self.lock.lock_file, contents)
+
+                # validate pid holding lock exists
+                if not pid_exists(int(contents[0])):
```

Comment:
I think you're asking when does the situation occur where a lock is held by a process that no longer exists. the answer to that question is that when a Daemon agent process is terminated or stopped during a scan or when performing a target action, potentially after a timeout expires. it could also happen if a resource agent action is halted before completing by killing the cli process.